lib/int64.js

```diff
@@ -0,0 +1,385 @@
+'use strict';
+
+const { Uint64 } = require('./uint64');
+
+const INT32_MAX = 0x7FFFFFFF;
+const UINT32_MAX = 0xFFFFFFFF;
+
+const charToNum = {
+  0: 0, 1: 1, 2: 2,
+  3: 3, 4: 4, 5: 5,
+  6: 6, 7: 7, 8: 8,
+  9: 9,
+  a: 10, A: 10,
+  b: 11, B: 11,
+  c: 12, C: 12,
+  d: 13, D: 13,
+  e: 14, E: 14,
+  f: 15, F: 15,
+};
+
+class Int64 {
+  constructor(value) {
+    this.value = new Uint32Array(2);
+    if (value instanceof Int64) {
+      this.value[0] = value.value[0];
+      this.value[1] = value.value[1];
+      return;
+    }
+    const numValue = Number(value);
+    if (Number.isNaN(numValue)) return;
+    if (numValue >= 0 && numValue <= INT32_MAX) {
+      this.value[0] = numValue;
+    } else if (numValue >= -INT32_MAX && numValue < 0) {
+      this.value[0] = numValue;
+      this.value[1] = UINT32_MAX;
+    } else if (numValue >= 0 && numValue <= Number.MAX_SAFE_INTEGER) {
+      this.value[0] = numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(numValue / (UINT32_MAX + 1));
+    } else if (numValue >= Number.MIN_SAFE_INTEGER && numValue < 0) {
+      this.value[0] = -numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(-numValue / (UINT32_MAX + 1));
+      this.value = Int64._conversion(this.value);
+    } else if (typeof value === 'string') {
+      const negative = value.startsWith('-');
+      if (negative) value = value.slice(1);
+
+      let radix = 10;
+      if (value.startsWith('0x')) {
+        radix = 16;
+      } else if (value.startsWith('0o')) {
+        radix = 8;
+      } else if (value.startsWith('0b')) {
+        radix = 2;
+      }
+      if (radix !== 10) value = value.slice(2);
+      const intRadix = new Int64(radix);
+      let res = Int64.zero();
+      for (let i = 0; i < value.length; i++) {
+        const digit = charToNum[value[i]];
+        const intDigit = new Int64(digit);
+        res = Int64.mult(res, intRadix);
+        if (negative) res.sub(intDigit);
+        else res.add(intDigit);
+      }
+      this.value[0] = res.value[0];
+      this.value[1] = res.value[1];
+    }
+  }
+
+  toInt32() {
```

Comment:
This method should just return `this.value[0]`.

Reply:
I thought that if the value is negative we should also return a negative number. For example, now it converts like that:
```js
new Int64('-1').toInt32() -> -1
```
If we just return `value[0]` it will be like this:
```js
new Int64('-1').toInt32() -> 4294967295
```
Do we need such type of behaviour?

Reply:
Yeah, I'm sorry, you should return `this.value[0] >> 0` here.
Separate method called `toUint32()` that returns just `this.value[0]` can be useful here, though.